src/jit/hwintrinsicxarch.cpp

```diff
@@ -1071,6 +1071,147 @@ GenTree* Compiler::impAvxOrAvx2Intrinsic(NamedIntrinsic        intrinsic,
 
     switch (intrinsic)
     {
+        case NI_AVX_Extract:
+        {
+            // Avx.Extract executes software implementation when the imm8 argument is not complie-time constant
+            assert(!mustExpand);
+
+            GenTree* lastOp   = impPopStack().val;
+            GenTree* vectorOp = impSIMDPopStack(TYP_SIMD32);
+            assert(lastOp->IsCnsIntOrI());
+            int ival          = (int)lastOp->AsIntCon()->IconValue();
+            baseType          = getBaseTypeOfSIMDType(info.compCompHnd->getArgClass(sig, sig->args));
+            var_types retType = JITtype2varType(sig->retType);
+            assert(varTypeIsArithmetic(baseType));
+
+            ival                            = ival & (32 / genTypeSize(baseType) - 1); // clear the unused bits
+            int            halfIndex        = 16 / genTypeSize(baseType);
```

Comment:
These two lines could be abstracted out for better readability (it's repeated below). It could also be made more efficient since we know that the type size is always a power of 2, though I don't think that's critical.
Both these suggestions can be left for future unless you are already going to do another commit on this PR, in which case you *might* consider extracting to a separate method.

Reply:
>  It could also be made more efficient since we know that the type size is always a power of 2

I would expect that the native compiler will properly recognize and transform `<pow-2-const> / value` and will transform it into the appropriate shift operation.

Reply:
> These two lines could be abstracted out for better readability 

Thanks, will change in this PR, and I will update the test cases to match https://github.com/dotnet/coreclr/pull/16957

Reply:
I'm not sure the compiler will do that, given that `genTypeSize` gets its value from a table, though perhaps it will be more clever than I think. In any case, I don't consider optimizing this a high-value investment.

Reply:
> I'm not sure the compiler will do that, given that genTypeSize gets its value from a table

You're probably right, I was thinking of the reverse operation `value / <pow-2-const>` 😄 

> I don't consider optimizing this a high-value investment.

Agreed